TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmode/BaseOpMode.java

```diff
@@ -77,10 +88,13 @@ protected void initHardware() {
     public void run() {
         super.run();
 
+        //bulk read set to auto
         List<LynxModule> allHubs = hardwareMap.getAll(LynxModule.class);
 
-        for (LynxModule module: allHubs) {
-            module.setBulkCachingMode(LynxModule.BulkCachingMode.AUTO);
+        if (useBulkRead) {
```

Comment:
this should not be in `run()`. the mode does not have to be repeatedly set, so move it to init please.